add_me_to_your_PATH/Dockerfile.template

```diff
@@ -3,17 +3,25 @@ ARG CUDA_VERSION
 
 # Extract Project.toml with pinned deps only + corresponding Manifest.toml
 
-FROM julia:${JULIA_VERSION}-buster as pinned-extractor
+FROM julia:${JULIA_VERSION}-buster as sysimage-project
 
-COPY *Project.toml *Manifest.toml .
+COPY Project.toml Manifest.toml ./julia_pod/sysimage.packages .
 
-# Remove all packages that are not pinned recursively from the manifest
 # (stdlibs count as pinned)
 RUN julia --project=. -e 'using Pkg;\
-                          unpinned = [p.name for p in values(Pkg.dependencies()) \
-                                      if p.is_direct_dep && !p.is_pinned && \
-                                      p.name ∉ values(Pkg.Types.stdlibs())]; \
-                          isempty(unpinned) || Pkg.rm(unpinned; mode=Pkg.PKGMODE_MANIFEST)'
+                          unregistered = [p.name for p in values(Pkg.dependencies()) \
+                                          if !p.is_tracking_registry]; \
+                          registered = [p.name for p in values(Pkg.dependencies()) \
+                                        if p.is_tracking_registry]; \
+                          previous = readlines("sysimage.packages"); \
+                          rm_deps = union(unregistered, setdiff(registered, previous)); \
+                          println("removing $rm_deps"); \
+                          isempty(rm_deps) || Pkg.rm(rm_deps; mode=Pkg.PKGMODE_MANIFEST); \
+                          direct = [p.name for p in values(Pkg.dependencies()) \
+                                          if !p.is_direct_dep]; \
+                          intersect!(rm_deps, direct); \
+                          isempty(rm_deps) || Pkg.rm(rm_deps; mode=Pkg.PKGMODE_PROJECT); \
+                          println(filter(contains(Regex(join(rm_deps, "|"))), readlines("Project.toml")))'
```

Comment:
This `println` doesn't actually update any files, right? Is that a bug?

Reply:
yeah I puzzled over this a while too.  it's the `Pkg.rm` that does th ework, the println is just for feedback/logs I think...

Reply:
ah ok!